pkg/controller/daemon/controller.go

```diff
@@ -406,11 +406,12 @@ func storeDaemonSetStatus(dsClient client.DaemonSetInterface, ds *extensions.Dae
 		return nil
 	}
 
+	ds.Status.DesiredNumberScheduled = desiredNumberScheduled
+	ds.Status.CurrentNumberScheduled = currentNumberScheduled
+	ds.Status.NumberMisscheduled = numberMisscheduled
+
 	var updateErr, getErr error
 	for i := 0; i <= StatusUpdateRetries; i++ {
-		ds.Status.DesiredNumberScheduled = desiredNumberScheduled
-		ds.Status.CurrentNumberScheduled = currentNumberScheduled
-		ds.Status.NumberMisscheduled = numberMisscheduled
```

Comment:
@liggitt You are right, I am sorry to not read the existing code carefully.Thanks for pointting out this. I will fix this soon in another PR.